File: server/order-service/src/main/java/com/sudo248/invoiceservice/controller/dto/ProductDto.java
package com.sudo248.invoiceservice.controller.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductDto implements Serializable {
    private String productId;
    private String name,description,sku;
    private List<ImageDto> images;
}

File: server/product-service/src/main/kotlin/com/sudoo/productservice/controller/dto/ProductDto.kt
package com.sudoo.productservice.controller.dto

import java.time.LocalDateTime

data class ProductDto(
    val productId: String? = null,
    val sku: String? = null,
    val name: String,
    val description: String,
    val price: Float,
    val listedPrice: Float,
    val amount: Int,
    val soldAmount: Int,
    val rate: Float,
    val discount: Int,
    val startDateDiscount: LocalDateTime,
    val endDateDiscount: LocalDateTime,
    val sellable: Boolean,
    val images: List<String> = emptyList(),
    val supplier: SupplierInfoDto? = null,
    val categories: List<CategoryInfoDto>? = null,
)